lib/util/connection.ts

```diff
@@ -51,6 +55,19 @@ export class Connection extends AsyncEventEmitter<ConnectionEvents> {
   private packetGroupReliability: PacketType = PacketType.ReliablePacket;
   unacknowledgedPackets: Map<number, number> = new Map();
   private TEMPDONTUSE: boolean = false;
+  private static internalSupportedVersions: ClientVersion[] = [
+    new ClientVersion(2020, 9, 7, 0),
+    new ClientVersion(2020, 10, 8, 0),
+    new ClientVersion(2020, 11, 17, 0),
+  ];
+  public static get supportedVersions(): ClientVersion[] {
```

Comment:
Not sure what the point of this is?

Reply:
To prevent the `internalSupportedVersions ` array from being modified

Reply:
you can use `public readonly supportedVersions` i have to refactor player.ts as well knowing this

Reply:
You can mark it as `private readonly` in that case

Reply:
(There may also be a case where you want to modify it, as this is a library I don't think we should be blocking end-users from necessarily doing that if they want to)

Reply:
Well given that the protocol is defined by the server I think the server should explicitly have control over which versions it supports

Reply:
> Well given that the protocol is defined by the server I think the server should explicitly have control over which versions it supports

Yes, that is fair.